### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by hjorthjort**
Documentation updates

### CODE REVIEW

File: x/staking/spec/delegation_sequence.puml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+alt delegating from an account (subtractTokens == true)
+    keeper -> keeper.bankKeeper : DelegateCoinsFromAccountToModule
+    keeper.bankKeeper -> DelegateCoinsFromAccountToModule **
+    DelegateCoinsFromAccountToModule -->  DelegateCoinsFromAccountToModule  : Check recipient has permission to stake
+    DelegateCoinsFromAccountToModule -> DelegateCoins **
+
+    DelegateCoins --> DelegateCoins : Check the delegator has enough balances of all tokens delegated
+    DelegateCoins --> DelegateCoins : Track delegation (register that it exists to keep track of it)
+    alt validator is currently bonded
```
Comment (shahankhatch) [inline x/staking/spec/delegation_sequence.puml]: Just noting that some of alts are mutually distinct choices, while some alts show a potential path based on a constraint check. So there's an opportunity to be more precise, if you think it's warranted.

File: x/staking/spec/delegation_sequence.puml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+title: Delegating (currently undelegated funds delegator)
+autonumber
+
+participant "msgServer (staking)"
+participant  "keeper (staking)" as keeper
+participant validator
+participant keeper.bankKeeper
+participant vestingAccount
+participant "func:SendCoins" as SendCoins
```
Comment (shahankhatch) [inline x/staking/spec/delegation_sequence.puml]: The entity with which the flow is interacting has changed on this line. The function is now like an entity and I think it's better to keep entities at the type/module abstraction, e.g., keeper, so it's easier to identify where the call is being directed (the function call name can be on the call's line).

**Review by shahankhatch — APPROVED** [review]

File: x/staking/spec/delegation_sequence.puml (hunk 2 above)
Comment (hjorthjort) [reply x/staking/spec/delegation_sequence.puml]: Fixed, using `group` instead to distinguish the actions that take place in a specific function

File: x/staking/spec/delegation_sequence.puml (hunk 1 above)
Comment (hjorthjort) [reply x/staking/spec/delegation_sequence.puml]: Done, using the `alt ... else` construct

**Pull request merged**